cloud/cluster/memory/subscription.go

```diff
@@ -0,0 +1,56 @@
+package memory
+
+import (
+	"github.com/scootdev/scoot/cloud/cluster"
+)
+
+// subscriber promptly reads its input and maintains a queue so that the sender
+// doesn't have to worry about a send blocking.
+type subscriber struct {
+	inCh    chan []cluster.NodeUpdate
+	outCh   chan []cluster.NodeUpdate
+	closeCh chan interface{}
+	queue   []cluster.NodeUpdate
+}
+
+func makeSubscription(initial []cluster.Node, closeCh chan interface{}, inCh chan []cluster.NodeUpdate) cluster.Subscription {
+	s := &subscriber{
+		inCh:    inCh,
+		outCh:   make(chan []cluster.NodeUpdate),
+		closeCh: closeCh,
+		queue:   nil,
+	}
+	go s.loop()
```

Comment:
This is an interface/doc issue primarily. If the expectation is that all users will read the impl then, yeah it's easy to understand, but it's unclear how this works just by looking at the functionality as seen in something like godoc.